### PR TITLE
Add interval datatype

### PR DESCRIPTION
**Pull request by chitang**
Add date_add function

### CODE REVIEW

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala [hunk 1]
```diff
@@ -568,6 +571,10 @@ object SoQLFunctions {
     NoDocs
   )
 
+  val TimeStampAdd = f("timestamp add", FunctionName("date_add"), Map("a" -> TimestampLike, "b" -> Set(SoQLInterval)), Seq(VariableType("a"), VariableType("b")), Seq.empty, VariableType("a"))(
```
Comment (rjmac) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:574]: ...`Map("a" -> TimestampLike), Seq(VariableType("a"), FixedType(SoQLInterval))`...

Comment (rjmac) [reply soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:574]: Also, any reason this isn't a `SpecialFunctions.Operator("+")` overload?  That should work fine...

**Review by rjmac — APPROVED** [review]
Could also add a conversion over in SoQLTypeInfo so that string literals which are formatted correctly are also interval literals, if you liked.

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala (hunk 1 above)
Comment (chitang) [reply soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:574]: I get what I want without Interval literals.  I will keep that minimal.

Comment (chitang) [reply soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:574]: i will try +

**Pull request merged**